### PR TITLE
[dataflow] Add an abstract level of BinaryOperation and UnaryOperation in Node hierarchy

### PR DESCRIPTION
**Pull request by CharlesZ-Chen**
Add abstract classes `UnaryOperation` and `BinaryOperation` to categorize operations Node types into `UnaryOperation` and `BinaryOperation`.

This could reduce code duplication in dataflow-framework, and also could enable us to have conciser code expression when developing dataflow analysis.

original PR #16 has some problems with travis, so I create this new one to test whether this time it would work.

### CODE REVIEW

File: dataflow/src/org/checkerframework/dataflow/cfg/node/UnaryOperationNode.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package org.checkerframework.dataflow.cfg.node;
+
+import com.sun.source.tree.UnaryTree;
+import java.util.Collection;
+import java.util.Collections;
+import org.checkerframework.javacutil.InternalUtils;
+
+public abstract class UnaryOperationNode extends Node {
+
+    protected UnaryTree tree;
```
Comment (wmdietl) [inline dataflow/src/org/checkerframework/dataflow/cfg/node/UnaryOperationNode.java]: Can these two fields be final?

File: dataflow/src/org/checkerframework/dataflow/cfg/node/BinaryOperationNode.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package org.checkerframework.dataflow.cfg.node;
+
+import com.sun.source.tree.BinaryTree;
+import java.util.Collection;
+import java.util.LinkedList;
+import org.checkerframework.javacutil.InternalUtils;
+
+public abstract class BinaryOperationNode extends Node {
+
+    protected BinaryTree tree;
```
Comment (wmdietl) [inline dataflow/src/org/checkerframework/dataflow/cfg/node/BinaryOperationNode.java]: Can these three fields be final?

File: dataflow/src/org/checkerframework/dataflow/cfg/node/BinaryOperationNode.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package org.checkerframework.dataflow.cfg.node;
+
+import com.sun.source.tree.BinaryTree;
+import java.util.Collection;
+import java.util.LinkedList;
+import org.checkerframework.javacutil.InternalUtils;
+
+public abstract class BinaryOperationNode extends Node {
```
Comment (wmdietl) [inline dataflow/src/org/checkerframework/dataflow/cfg/node/BinaryOperationNode.java]: Add a simple javadoc comment here and in UnaryOperationNode.

**Comment by CharlesZ-Chen** [comment]
Already made a PR against to `typetools`:
https://github.com/typetools/checker-framework/pull/968

Will close this one once the upstream PR has been approved and merged.

**Comment by CharlesZ-Chen** [comment]
upstream PR https://github.com/typetools/checker-framework/pull/968 merged. Thus close this one.

**Pull request closed without merging**